packages/api/src/queueAdapters/base.ts

```diff
@@ -56,6 +58,8 @@ export abstract class BaseAdapter {
 
   public abstract getJobLogs(id: string): Promise<string[]>;
 
+  public abstract getMetrics(type: JobRetryStatus, start?: number, end?: number): Promise<QueueMetrics | undefined | null>;
```

Comment:
Why did you used the `JobRetryStatus` instead of `JobStatus`

Reply:
From what I gathered, only valid status for metrics is `completed` and `failed`. I can also create a new type for this

Reply:
So it is better to create a new type for it